hocon/src/main/scala/com/avsystem/commons/hocon/ConfigCompanion.scala

```diff
@@ -0,0 +1,69 @@
+package com.avsystem.commons
+package hocon
+
+import com.avsystem.commons.meta.MacroInstances
+import com.avsystem.commons.misc.ValueOf
+import com.avsystem.commons.serialization.{GenCodec, GenKeyCodec, GenObjectCodec, HasGenObjectCodecWithDeps}
+import com.avsystem.commons.serialization.GenCodec.ReadFailure
+import com.typesafe.config.{Config, ConfigFactory, ConfigObject}
+
+import scala.concurrent.duration.*
+
+trait CommonsHoconCodecs {
+  implicit final val configCodec: GenCodec[Config] = GenCodec.nullable(
+    input =>
+      input.readCustom(ConfigValueMarker).map {
+        case obj: ConfigObject => obj.toConfig
+        case v => throw new ReadFailure(s"expected a config OBJECT, got ${v.valueType}")
+      }.getOrElse {
+        ConfigFactory.parseString(input.readSimple().readString())
+      },
```

Comment:
nit: I could be convinced this is one of the rare instances where `.fold` looks cleaner

Reply:
Refactored to fold